xformers/csrc/attention/cuda/fmha/kernel_backward.h

```diff
@@ -1603,6 +1606,21 @@ struct AttentionBackwardKernel {
             [&](int accum_n) {});
       }
 
+      if (p.use_alibi) {
+        // apply bias from Alibi embeddings
+        auto lane_offset = MatmulQK::AccumLambdaIterator::get_lane_offset(
+            lane_id, warp_id, output_tile_coords);
+        MatmulQK::AccumLambdaIterator::iterateRows(
+            lane_offset,
+            [&](int accum_n) {},
+            [&](int accum_m, int accum_n, int idx) {
+              // remember we are transposed
+              accum[idx] += alibi_base * (static_cast<float>(key_start) + static_cast<float>(accum_m) - static_cast<float>(query_start) - static_cast<float>(accum_n));
```

Comment:
```suggestion
              accum[idx] += alibi_base * static_cast<float>(key_start + accum_m - query_start - accum_n);
```

Reply:
Or is there a hidden meaning in the individual casts?

Reply:
Some of them are unsigned, some signed. It works incorrectly with the code you suggest, I guess we can instead cast all of them to the same integer type but I don't know if it is faster

Reply:
Integers always work faster than floats. Let's cast to `int` then.